src/edgepi/adc/adc_multiplexers.py

```diff
@@ -27,8 +26,7 @@ def _format_mux_values(mux_p: CH, mux_n: CH):
 
     return mux_p_val, mux_n_val, mask
 
-
-def generate_mux_opcodes(mux_updates: dict):
+def generate_mux_opcodes(adc1_reg, adc1_mux, adc2_reg, adc2_mux):
```

Comment:
pack is a surprisingly slow function, so replacing it improved performance a lot. The logging and dictionary operations also slowed this function down a reasonable amount, which is why I removed them. I tried to avoid removing logging statements where possible, however they did have an effect on the performance.